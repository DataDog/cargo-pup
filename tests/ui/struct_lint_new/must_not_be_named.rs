//@compile-flags: --crate-name test_must_not_be_named
//@compile-flags: --crate-type lib

// This test verifies that the StructRule::MustNotBeNamed rule works correctly by
// flagging structs that match the forbidden naming pattern "Bad*".

// These structs match the forbidden pattern, should trigger errors
pub struct BadName { //~ ERROR: Struct must not match pattern 'Bad*'
    field: i32,
}

pub struct BadStruct { //~ ERROR: Struct must not match pattern 'Bad*'
    value: String,
}

// These structs don't match the forbidden pattern, shouldn't trigger errors
pub struct GoodName {
    data: Vec<u8>,
}

pub struct AcceptableName {
    id: u64,
    name: String,
} 