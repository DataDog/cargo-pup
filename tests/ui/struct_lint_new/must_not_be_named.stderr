error: Struct must not match pattern 'Bad*'
  --> tests/ui/struct_lint_new/must_not_be_named.rs:10:1
   |
LL | pub struct BadName {
   | ^^^^^^^^^^^^^^^^^^^^
   |
   = help: Choose a name that doesn't match this pattern
   = note: Applied by cargo-pup rule 'must_not_be_named_test'.
   = note: `#[deny(struct_lint_must_not_be_named)]` on by default

error: Struct must not match pattern 'Bad*'
  --> tests/ui/struct_lint_new/must_not_be_named.rs:14:1
   |
LL | pub struct BadStruct {
   | ^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Choose a name that doesn't match this pattern
   = note: Applied by cargo-pup rule 'must_not_be_named_test'.

error: aborting due to 2 previous errors

